vulkano/src/memory/device_memory.rs

```
@@ -1427,6 +1456,94 @@ impl MemoryMapInfo {
             }));
         }
 
+        if let Some(placed_address) = placed_address {
+            if !memory.device.enabled_extensions().ext_map_memory_placed {
+                return Err(Box::new(ValidationError {
+                    context: "placed_address".into(),
+                    problem: "is not empty".into(),
+                    requires_one_of: RequiresOneOf(&[RequiresAllOf(&[Requires::DeviceExtension(
+                        "ext_map_memory_placed",
+                    )])]),
+                    ..Default::default()
+                }));
+            }
+
+            let features = memory.device.enabled_features();
+            if !features.memory_map_placed {
+                return Err(Box::new(ValidationError {
+                    context: "placed_address".into(),
+                    problem: "is not empty".into(),
+                    requires_one_of: RequiresOneOf(&[RequiresAllOf(&[Requires::DeviceFeature(
+                        "memory_map_placed",
+                    )])]),
+                    vuids: &["VUID-VkMemoryMapInfoKHR-flags-09569"],
```

Is this the right vuid to use here? https://registry.khronos.org/vulkan/specs/1.3-extensions/man/html/VkMemoryMapInfoKHR.html#VUID-VkMemoryMapInfoKHR-flags-09569

Its not exactly covering the same condition. Should I leave the VUID out instead?

---

It looks like the right VUID to me. What are you unsure about?

---

If it's redundant, we don't need the flag in Vulkano, as long as we add it before calling Vulkan. Which is what's done here.

---

Yes, exactly. The VUID here is targeting the flag while the error check is regarding whether `placed_address` is provided or not. The presence of `placed_address` in vulkan implies the flag.

I was having a hard time coming up with an API which doesn't have the implicit behavior. I guess we could make a new fn with its own set of arguments instead? Let me know if this is a dealbreaker.

---

I'm not that attached to it, and you're right that following Vulkan to the letter makes Vulkano more robust against future additions.

---

Great, I will update this in the next few hours.

---

I've updated the PR. One more possible interface to consider is instead of adding fields to `MemoryMapInfo` a new `map_placed` function with a separate `MemoryMapPlacedInfo` could be created which implies `MemoryMapFlags::PLACED_EXT`. This interface has less parameter combinations which work unexpectedly. For example, as is if there is no flags and a placed address specified, it will silently fail. I don't feel strongly either way, I will defer the decision to @marc0246 and @Rua 🙏

---

If we have more cycles of reviewing it might be worthwhile all of us finding a time and sitting down together to hammer out an ideal api interface here. I think it will be faster for all parties involved to meet synchronously for 15m than more review cycles.

It seems like we're all active in U.S. Central mornings, happy to coordinate a time if needed.

---

Ahh, I see. I've moved it.